hookman/hooks.py

```diff
@@ -56,92 +60,78 @@ class HookMan:
 
     def __init__(self, *, specs: HooksSpecs, plugin_dirs: List[Path]):
         self.specs = specs
-
-        config_files = []
-        for plugin_dir in plugin_dirs:
-            config_files += plugin_dir.glob('**/plugin.yaml')
-
-        if config_files:
-            self.plugin_config_files = config_files
-        else:
-            raise FileNotFoundError("The given path doesn't have a .yaml file")
-
+        self.plugins_dirs = plugin_dirs
         self.hooks_available = {
             f'{hook.__name__.lower()}': f'{specs.project_name.lower()}_v{specs.version}_{hook.__name__.lower()}'
             for hook in specs.hooks
         }
 
-    def get_hook_caller(self) -> 'HookCaller':
+    def install_plugin(self, plugin_file_path: Path, dst_path: Path):
+        """
+        Extract the content of the zip file into plugin_dirs
+        """
+        plugin_file_zip = ZipFile(plugin_file_path)
+        PluginInfo.plugin_file_validation(plugin_file_zip=plugin_file_zip)
+
+        if dst_path not in self.plugins_dirs:
+            raise InvalidDestinationPathError(f"Invalid destination path, {dst_path} is not one of "
+                                              f"the paths that were informed when the HookMan "
+                                              f"object was initialized: {self.plugins_dirs}.")
+
+        plugin_name = Path(plugin_file_zip.filename).resolve().stem
+        plugins_dirs = [x for x in dst_path.iterdir() if x.is_dir()]
+
+        if plugin_name in [x.name for x in plugins_dirs]:
+            raise PluginAlreadyInstalledError("Plugin already installed")
+
+        plugin_destination_folder = dst_path / plugin_name
+        plugin_destination_folder.mkdir(parents=True)
+        plugin_file_zip.extractall(plugin_destination_folder)
+
+    def remove_plugin(self, plugin_name: str):
+        for plugin in self.plugins_available():
+            if plugin.name == plugin_name:
+                shutil.rmtree(plugin.location.parent)
+                break
+
+
+    def plugins_available(self) -> Optional[List[PluginInfo]]:
+        """
+        Return a list with all plugins that are available on the plugins_dirs.
+        The list contains a PluginInfo object that contains information about the plugin
+        (configuration available at the YAML file and computed values )
+        """
+        plugins_available = []
+        plugin_config_files = hookman_utils.find_config_files(self.plugins_dirs)
+        if plugin_config_files:
```

Comment:
`return [PluginInfo(x) for x in plugin_config_files]`? For this to work `find_config_files` needs to always return a list, of course.

Reply:
`find_config_files` always return a list, in cases that there aren't plugins, an empty list is returned ;)

-[x] Change to `list comprehension`